server/src/main/java/org/opensearch/action/admin/cluster/stats/TransportClusterStatsAction.java

```diff
@@ -124,14 +134,27 @@ protected ClusterStatsResponse newResponse(
                 + " the cluster state that are too slow for a transport thread"
         );
         ClusterState state = clusterService.state();
-        return new ClusterStatsResponse(
-            System.currentTimeMillis(),
-            state.metadata().clusterUUID(),
-            clusterService.getClusterName(),
-            responses,
-            failures,
-            state
-        );
+        if (request.applyMetricFiltering()) {
+            return new ClusterStatsResponse(
+                System.currentTimeMillis(),
+                state.metadata().clusterUUID(),
+                clusterService.getClusterName(),
+                responses,
+                failures,
+                state,
+                request.requestedMetrics(),
+                request.indicesMetrics()
```

Comment:
Why are we sending the request back in response ? This seems to be unnecessary ser/deser

Reply:
These metrics are not sent back, they are only used in the response constructor to check if node/indices stats need to be computed or not.